src/ray/pubsub/subscriber.cc

```diff
@@ -219,6 +219,15 @@ std::string SubscriberChannel::DebugString() const {
 /// Subscriber
 ///////////////////////////////////////////////////////////////////////////////
 
+Subscriber::~Subscriber() {
+  if (!mutex_.LockWhenWithTimeout(absl::Condition(this, &Subscriber::CheckNoLeaks_Locked),
```

Comment:
Should we do this only when debug mode is on?

Reply:
Removing this for now. In future it will be a debug only check.